spring-cloud-azure-context/src/main/java/com/microsoft/azure/spring/cloud/context/core/AzureAdmin.java

```diff
@@ -207,27 +232,50 @@ private ServiceBusSubscription createServiceBusTopicSubscription(Tuple<Topic, St
                                        .create();
     }
 
-    public RedisCache getRedisCache(String name) {
+    private RedisCache getRedisCache(String name) {
         return azure.redisCaches().getByResourceGroup(resourceGroup, name);
     }
 
-    public RedisCache createRedisCache(String name) {
+    private RedisCache createRedisCache(String name) {
         return azure.redisCaches().define(name).withRegion(region).withExistingResourceGroup(resourceGroup)
                     .withBasicSku().create();
     }
 
     public RedisCache getOrCreateRedisCache(String name) {
-        return getOrCreate(this::getRedisCache, this::createRedisCache).apply(name);
+        return getOrCreate(this::getRedisCache, this::createRedisCache, RedisCache.class).apply(name);
     }
 
-    private <T, R> Function<T, R> getOrCreate(Function<T, R> getter, Function<T, R> creator) {
+    private <T, R> Function<T, R> getOrCreate(Function<T, R> getter, Function<T, R> creator, Class<R> resourceType) {
         return t -> {
             R result = getter.apply(t);
             if (result != null) {
                 return result;
             }
 
-            return creator.apply(t);
+            return logCreator(creator, resourceType).apply(t);
+        };
+    }
+
+    private <T, R> Function<T, R> logCreator(Function<T, R> creator, Class<R> resourceType) {
```

Comment:
Could u please help to try another name rather than `logCreator`? I am not sure if I understand correctly, here we do resource creation, and add some log here. The name `logCreator` looks it will create a log.